src/main/java/de/terrestris/momo/security/MomoAuthenticationProvider.java

```diff
@@ -0,0 +1,192 @@
+package de.terrestris.momo.security;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.authentication.DisabledException;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.transaction.annotation.Transactional;
+
+import de.terrestris.momo.dao.UserGroupRoleDao;
+import de.terrestris.momo.model.security.UserGroupRole;
+import de.terrestris.shogun2.dao.UserDao;
+import de.terrestris.shogun2.model.Role;
+import de.terrestris.shogun2.model.User;
+import de.terrestris.shogun2.security.Shogun2AuthenticationProvider;
+
+/**
+ *
+ * @author Daniel Koch
+ * @author terrestris GmbH & Co. KG
+ *
+ */
+public class MomoAuthenticationProvider extends Shogun2AuthenticationProvider {
+
+	/**
+	 * The Logger
+	 */
+	private static final Logger LOG = Logger
+			.getLogger(MomoAuthenticationProvider.class);
+
+	/**
+	 *
+	 */
+	@Autowired
+	private UserDao<User> userDao;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private UserGroupRoleDao<UserGroupRole> userGroupRoleDao;
+
+	/**
+	 *
+	 */
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
```

Comment:
As the parent class `Shogun2AuthenticationProvider` has this field, too, why don't you remove this field here and call `getPasswordEncoder()` where you need it?